services/qs-spark-service/src/main/java/ai/quantumics/api/controller/RuleController.java

```diff
@@ -0,0 +1,68 @@
+package ai.quantumics.api.controller;
+
+import ai.quantumics.api.service.RuleService;
+import ai.quantumics.api.vo.RuleDetails;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import springfox.documentation.spring.web.json.Json;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/v1/qsrules")
+@Api(value = "QuantumSpark Service API ")
+public class RuleController {
+
+    private final RuleService ruleService;
+
+    public RuleController(
+            RuleService ruleServiceCi) {
+        ruleService = ruleServiceCi;
+    }
+
+    @ApiOperation(value = "rules", response = Json.class)
+    @PostMapping("/{projectId}")
+    @ApiResponses(
+            value = {
+                    @ApiResponse(code = 200, message = "OKAY"),
+                    @ApiResponse(code = 409, message = "Failure message")
+            })
+    public ResponseEntity<Object> saveRule(
+            @RequestBody final RuleDetails ruleDetails,
+            @PathVariable(value = "projectId") final int projectId)
```

Comment:
userid and project id should be passed and validated whether it exists. it is applicable to methods.. so that you will get data for the respective one